fontforge/fvfonts.c

```diff
@@ -1280,35 +1283,138 @@ static RefChar *InterpRefs(RefChar *base, RefChar *other, real amount, SplineCha
-		    (test->unicode_enc!=-1 || strcmp(test->sc->name,base->sc->name)==0 ) )
+		    (test->unicode_enc!=-1 || strcmp(test->sc->name,base->sc->name)==0 ) ) {
 	break;
 	}
+	}
 	if ( test!=NULL ) {
 	    test->checked = true;
-	    cur = RefCharCreate();
-	    free(cur->layers);
-	    *cur = *base;
-	    cur->orig_pos = cur->sc->orig_pos;
-	    for ( i=0; i<6; ++i )
-		cur->transform[i] = base->transform[i] + amount*(other->transform[i]-base->transform[i]);
-	    cur->layers = NULL;
-	    cur->layer_cnt = 0;
-	    cur->checked = false;
-	    if ( head==NULL )
-		head = cur;
-	    else
-		last->next = cur;
-	    last = cur;
-	} else
-	    LogError( _("In character %s, could not find reference to %s\n"),
-		    sc->name, base->sc->name );
+	    if ( !check_only ) {
+		cur = RefCharCreate();
+		free(cur->layers);
+		*cur = *base;
+		cur->orig_pos = cur->sc->orig_pos;
+		for ( i=0; i<6; ++i )
+			cur->transform[i] = base->transform[i] + amount*(other->transform[i]-base->transform[i]);
+		cur->layers = NULL;
+		cur->layer_cnt = 0;
+		cur->checked = false;
+		if ( head==NULL )
+			head = cur;
+		else
+			last->next = cur;
+		last = cur;
+	    }
+	} else {
+	    LogError( _("Glyph %s: could not find reference to %s\n"),
+		    glyphname, base->sc->name );
+	    if ( check_only ) return ( NULL );
+	}
 	base = base->next;
 	if ( test==other && other!=NULL )
 	    other = other->next;
     }
+
+// If check_only, doesn't matter as long as it's not NULL, as if using check_only only NULL/non-NULL matters
+if ( check_only ) 
+    return ( temp );
 return( head );
 }
 
+/* In variable fonts, and MM fonts before them, "compatibility" between
+ * contours is needed to acheive a good (or any) result.
+ *
+ * FontForge has no such limitation, and will force contours to be compatible,
+ * then interpolate those. Usually, the result looks horrible.
+ *
+ * So, since the user is probably using this feature to test variable fonts,
+ * ask them if they even want FontForge to force compatibility. */
+bool InterpolationSanity(SplineSet *base, SplineSet *other, int order2, RefChar* brc, RefChar* orc, char* glyphname) {
+    SplineSet *bss=NULL, *oss=NULL;
+    char* prefix;
+    int bmcc = 0, omcc = 0;
+    bool ret = true;
+
+    // Count contours...
+    for ( bss = base; bss != NULL; bss=bss->next ) bmcc++;
+    for ( oss = other; oss != NULL; oss=oss->next ) omcc++;
+
+    if (glyphname == NULL) {
+        prefix = "";
+    } else {
+        prefix = smprintf(_("Glyph %s: "), glyphname);
+    }
+
+    if ( brc != NULL || orc != NULL ) {
+        RefChar *temp = InterpRefs(brc, orc, 0.5, glyphname, true);
+        ret = temp != NULL;
+    }
+
+    if (bmcc != omcc) {
+        LogError(_("%sIncompatible contour counts: %d vs %d."), prefix, bmcc, omcc);
+        ret = false;
+    }
+
+    SplinePoint* sp;
+    int contour_count = 1;
+    for ( bss = base, oss = other; bss!=NULL; bss=bss->next, oss=oss->next, contour_count++ ) {
+        if (!ret) break;
```

Comment:
I may be in a minority but I prefer `ret && bss!=NULL` as the loop test rather than adding additional flow control inside the loop (given that the `break` test comes first anyway).